operators/src/adapters/sparse_tiles_fill_adapter.rs

```diff
@@ -0,0 +1,696 @@
+use std::{pin::Pin, task::Poll};
+
+use futures::{ready, Stream};
+use geoengine_datatypes::{
+    primitives::TimeInterval,
+    raster::{
+        EmptyGrid2D, GeoTransform, GridBoundingBox2D, GridBounds, GridIdx2D, GridShape2D, GridStep,
+        Pixel, RasterTile2D,
+    },
+    util::Result,
+};
+use pin_project::pin_project;
+
+#[derive(Debug, PartialEq, Clone, Copy)]
+enum StateInner {
```

Comment:
can we find more speaking names for the states?

Reply:
i created a new version where all transitions are in one block 467427a